cmd/promtool/rules.go

```diff
@@ -92,22 +94,70 @@ func (importer *ruleImporter) importAll(ctx context.Context) (errs []error) {
 	return errs
 }
 
+type cacheEntry struct {
+	ts   time.Time
+	lset labels.Labels
+}
+
+// stalenessCache tracks staleness of series between blocks of data.
+type stalenessCache struct {
+	step time.Duration
+	// seriesCur store the labels of series that were seen previous block.
+	seriesPrev map[uint64]*cacheEntry
+}
+
+func newStalenessCache(step time.Duration) *stalenessCache {
+	return &stalenessCache{
+		step:       step,
+		seriesPrev: map[uint64]*cacheEntry{},
+	}
+}
+
+func (c *stalenessCache) trackStaleness(ts time.Time, lset labels.Labels) {
```

Comment:
What about changing `lset labels.Labels` param to `hash uint64` directly? So that `lset.Hash()` doesn't have to be calculated multiple times.
Same for `isStale` and `remove` methods.

Reply:
good idea. I will make that change. thanks!